system/security/cryptohelper/pke.cpp

```diff
@@ -113,6 +113,49 @@ class CLoadedPublicKeyFromMemory : public CLoadedKey
     }
 };
 
+class CLoadedPublicKeyFromCertFile : public CLoadedKey
+{
+public:
+    CLoadedPublicKeyFromCertFile(const char *certFile)
+    {
+        if (!loadKeyFromFile(certFile))
+            throwEVPExceptionV(0, "CLoadedPublicKeyFromCertFile: failed to open cert: %s", certFile);
+
+        OwnedX509 cert = PEM_read_bio_X509(keyBio, nullptr, nullptr, nullptr);
+        if (!cert)
+            throwEVPExceptionV(0, "Failed to read certificate: %s", certFile);
+        OwnedEVPPkey pubKey = X509_get_pubkey(cert);
+        if (!pubKey)
+            throwEVPExceptionV(0, "Failed to get public key from certificate key: %s", certFile);
```

Comment:
Same picky comment here: "Failed to get key from public certificate".

Reply:
will change message